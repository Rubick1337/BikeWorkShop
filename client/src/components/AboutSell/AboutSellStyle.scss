.img_dilevery
{
  background-image: url(../../images/7a42f224198bf7b519128db53370dab8.gif);
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
  display: flex;
  align-items: center;
  width: 223px;
  height: 125px;
  justify-content: center;
  flex-shrink: 0;
}
.img_garant
{
  background-image: url(../../images/4ee3f524200ff0bbf5791ee2a73cd0d9.gif);
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
  display: flex;
  align-items: center;
  width: 223px;
  height: 125px;
  justify-content: center;
  flex-shrink: 0;
}
.img_safe
{
  background-image: url(../../images/0f12e5287209a7e7dd3df101a6e8522b.gif);
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
  display: flex;
  align-items: center;
  width: 223px;
  height: 125px;
  justify-content: center;
  flex-shrink: 0;
}
.img_help
{
  background-image: url(../../images/6a3827516680631e900f5ca5ae311816.gif);
  background-repeat: no-repeat;
  background-position: center;
  background-size: contain;
  display: flex;
  align-items: center;
  width: 223px;
  height: 125px;
  justify-content: center;
  flex-shrink: 0;
}
.container__item__about
{
  display: flex;
  width: 223px;
  height: 278px;
  flex-direction: column;
  align-items: flex-start;
  gap: 15px;
}
.title_item__about h3
{
  color: #121A0F;
  font-feature-settings: 'dlig' on;
  font-family: Manrope;
  font-size: 20px;
  font-style: normal;
  font-weight: 500;
  line-height: 24px; /* 120% */
}
.text__item__about h4
{
  color: #275108;
  font-feature-settings: 'dlig' on;
  font-family: Manrope;
  font-size: 12px;
  font-style: normal;
  font-weight: 400;
  line-height: 21px; /* 175% */
}
.all__item
{
  display: flex;
  padding: 16px;
  align-items: flex-start;
  align-content: flex-start;
  gap: 39px;
  flex-wrap: wrap;
  justify-content: center;
}
.container__about__sell
{
  display: flex;
  flex-direction: column;
  align-items: center;
}