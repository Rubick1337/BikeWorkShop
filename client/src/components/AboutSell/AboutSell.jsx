import './AboutSellStyle.scss'

export default function AboutSell() {
    return (
        <div className="container__about__sell">
            <div className="title__about">
                <h2>О наших товарах</h2>
            </div>
            <div className="all__item">
                <div className="container__item__about">
                    <div className="dilevery__container">
                        <div className="img_dilevery"></div>
                        <div className="title_item__about">
                            <h3>Партнерские поставки</h3>
                        </div>
                        <div className="text__item__about">
                            <h4>Наши велосипеды и комплектующие поставляются только от проверенных партнеров,
                                что гарантирует высокое качество продукции.</h4>
                        </div>

                    </div>
                </div>
                <div className="container__item__about">
                    <div className="dilevery__container">
                        <div className="img_garant"></div>
                        <div className="title_item__about">
                            <h3>Гарантия качества</h3>
                        </div>
                        <div className="text__item__about">
                            <h4>Каждая деталь и велосипед проходят строгий контроль качества
                                и сопровождаются гарантией на срок до 2 лет.</h4>
                        </div>

                    </div>
                </div>
                <div className="container__item__about">
                    <div className="dilevery__container">
                        <div className="img_safe"></div>
                        <div className="title_item__about">
                            <h3>Надежность</h3>
                        </div>
                        <div className="text__item__about">
                            <h4>Мы выбираем материалы, которые обеспечивают долговечность
                                и надежность вашего велосипеда даже при интенсивном использовании.</h4>
                        </div>
                    </div>
                </div>
                <div className="container__item__about">
                    <div className="dilevery__container">
                        <div className="img_help"></div>
                        <div className="title_item__about">
                            <h3>Поддержка клиентов</h3>
                        </div>
                        <div className="text__item__about">
                            <h4>Покупая у нас, вы получаете квалифицированную поддержку и
                                возможность сервисного обслуживания у наших партнеров.</h4>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
}
