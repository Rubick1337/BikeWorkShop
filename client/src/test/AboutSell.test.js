import React from 'react';
import { render } from '@testing-library/react';
import AboutSell from '../components/AboutSell/AboutSell';

describe('AboutSell Component', () => {
    // Тест на сравнение с снимком
    it('renders correctly', () => {
        const { container } = render(<AboutSell />);
        expect(container).toMatchSnapshot();
    });

    it('renders correct headings', () => {
        const { getByText } = render(<AboutSell />);
        expect(getByText('О наших товарах')).toBeInTheDocument();
        expect(getByText('Партнерские поставки')).toBeInTheDocument();
        expect(getByText('Гарантия качества')).toBeInTheDocument();
        expect(getByText('Надежность')).toBeInTheDocument();
        expect(getByText('Поддержка клиентов')).toBeInTheDocument();
    });

    it('renders correct descriptions', () => {
        const { getByText } = render(<AboutSell />);
        expect(getByText(/наши велосипеды и комплектующие поставляются только от проверенных партнеров/i)).toBeInTheDocument();
        expect(getByText(/каждая деталь и велосипед проходят строгий контроль качества и сопровождаются гарантией на срок до 2 лет/i)).toBeInTheDocument();
        expect(getByText(/мы выбираем материалы, которые обеспечивают долговечность и надежность вашего велосипеда даже при интенсивном использовании/i)).toBeInTheDocument();
        expect(getByText(/покупая у нас, вы получаете квалифицированную поддержку и возможность сервисного обслуживания у наших партнеров/i)).toBeInTheDocument();
    });

    it('renders without crashing', () => {
        const { getByText } = render(<AboutSell />);
        expect(getByText('О наших товарах')).toBeInTheDocument();
    });

    it('handles unexpected props gracefully', () => {
        const { container } = render(<AboutSell unexpectedProp="unexpected" />);
        expect(container).toBeInTheDocument();
    });
});