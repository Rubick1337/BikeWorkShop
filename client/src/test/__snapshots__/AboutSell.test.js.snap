// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutSell Component handles unexpected props gracefully 1`] = `
<div>
  <div
    class="container__about__sell"
  >
    <div
      class="title__about"
    >
      <h2>
        О наших товарах
      </h2>
    </div>
    <div
      class="all__item"
    >
      <div
        class="container__item__about"
      >
        <div
          class="dilevery__container"
        >
          <div
            class="img_dilevery"
          />
          <div
            class="title_item__about"
          >
            <h3>
              Партнерские поставки
            </h3>
          </div>
          <div
            class="text__item__about"
          >
            <h4>
              Наши велосипеды и комплектующие поставляются только от проверенных партнеров, что гарантирует высокое качество продукции.
            </h4>
          </div>
        </div>
      </div>
      <div
        class="container__item__about"
      >
        <div
          class="dilevery__container"
        >
          <div
            class="img_garant"
          />
          <div
            class="title_item__about"
          >
            <h3>
              Гарантия качества
            </h3>
          </div>
          <div
            class="text__item__about"
          >
            <h4>
              Каждая деталь и велосипед проходят строгий контроль качества и сопровождаются гарантией на срок до 2 лет.
            </h4>
          </div>
        </div>
      </div>
      <div
        class="container__item__about"
      >
        <div
          class="dilevery__container"
        >
          <div
            class="img_safe"
          />
          <div
            class="title_item__about"
          >
            <h3>
              Надежность
            </h3>
          </div>
          <div
            class="text__item__about"
          >
            <h4>
              Мы выбираем материалы, которые обеспечивают долговечность и надежность вашего велосипеда даже при интенсивном использовании.
            </h4>
          </div>
        </div>
      </div>
      <div
        class="container__item__about"
      >
        <div
          class="dilevery__container"
        >
          <div
            class="img_help"
          />
          <div
            class="title_item__about"
          >
            <h3>
              Поддержка клиентов
            </h3>
          </div>
          <div
            class="text__item__about"
          >
            <h4>
              Покупая у нас, вы получаете квалифицированную поддержку и возможность сервисного обслуживания у наших партнеров.
            </h4>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AboutSell Component renders correctly 1`] = `
<div>
  <div
    class="container__about__sell"
  >
    <div
      class="title__about"
    >
      <h2>
        О наших товарах
      </h2>
    </div>
    <div
      class="all__item"
    >
      <div
        class="container__item__about"
      >
        <div
          class="dilevery__container"
        >
          <div
            class="img_dilevery"
          />
          <div
            class="title_item__about"
          >
            <h3>
              Партнерские поставки
            </h3>
          </div>
          <div
            class="text__item__about"
          >
            <h4>
              Наши велосипеды и комплектующие поставляются только от проверенных партнеров, что гарантирует высокое качество продукции.
            </h4>
          </div>
        </div>
      </div>
      <div
        class="container__item__about"
      >
        <div
          class="dilevery__container"
        >
          <div
            class="img_garant"
          />
          <div
            class="title_item__about"
          >
            <h3>
              Гарантия качества
            </h3>
          </div>
          <div
            class="text__item__about"
          >
            <h4>
              Каждая деталь и велосипед проходят строгий контроль качества и сопровождаются гарантией на срок до 2 лет.
            </h4>
          </div>
        </div>
      </div>
      <div
        class="container__item__about"
      >
        <div
          class="dilevery__container"
        >
          <div
            class="img_safe"
          />
          <div
            class="title_item__about"
          >
            <h3>
              Надежность
            </h3>
          </div>
          <div
            class="text__item__about"
          >
            <h4>
              Мы выбираем материалы, которые обеспечивают долговечность и надежность вашего велосипеда даже при интенсивном использовании.
            </h4>
          </div>
        </div>
      </div>
      <div
        class="container__item__about"
      >
        <div
          class="dilevery__container"
        >
          <div
            class="img_help"
          />
          <div
            class="title_item__about"
          >
            <h3>
              Поддержка клиентов
            </h3>
          </div>
          <div
            class="text__item__about"
          >
            <h4>
              Покупая у нас, вы получаете квалифицированную поддержку и возможность сервисного обслуживания у наших партнеров.
            </h4>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AboutSell Component renders correctly with different props 1`] = `
<div>
  <div
    class="container__about__sell"
  >
    <div
      class="title__about"
    >
      <h2>
        О наших товарах
      </h2>
    </div>
    <div
      class="all__item"
    >
      <div
        class="container__item__about"
      >
        <div
          class="dilevery__container"
        >
          <div
            class="img_dilevery"
          />
          <div
            class="title_item__about"
          >
            <h3>
              Партнерские поставки
            </h3>
          </div>
          <div
            class="text__item__about"
          >
            <h4>
              Наши велосипеды и комплектующие поставляются только от проверенных партнеров, что гарантирует высокое качество продукции.
            </h4>
          </div>
        </div>
      </div>
      <div
        class="container__item__about"
      >
        <div
          class="dilevery__container"
        >
          <div
            class="img_garant"
          />
          <div
            class="title_item__about"
          >
            <h3>
              Гарантия качества
            </h3>
          </div>
          <div
            class="text__item__about"
          >
            <h4>
              Каждая деталь и велосипед проходят строгий контроль качества и сопровождаются гарантией на срок до 2 лет.
            </h4>
          </div>
        </div>
      </div>
      <div
        class="container__item__about"
      >
        <div
          class="dilevery__container"
        >
          <div
            class="img_safe"
          />
          <div
            class="title_item__about"
          >
            <h3>
              Надежность
            </h3>
          </div>
          <div
            class="text__item__about"
          >
            <h4>
              Мы выбираем материалы, которые обеспечивают долговечность и надежность вашего велосипеда даже при интенсивном использовании.
            </h4>
          </div>
        </div>
      </div>
      <div
        class="container__item__about"
      >
        <div
          class="dilevery__container"
        >
          <div
            class="img_help"
          />
          <div
            class="title_item__about"
          >
            <h3>
              Поддержка клиентов
            </h3>
          </div>
          <div
            class="text__item__about"
          >
            <h4>
              Покупая у нас, вы получаете квалифицированную поддержку и возможность сервисного обслуживания у наших партнеров.
            </h4>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
